File: 413_Arithmetic_Slices.py
class Solution:
    def numberOfArithmeticSlices(self, A: 'List[int]') -> 'int':
        dp = res = 0
        for i in range(len(A) - 2):
            if A[i + 1] - A[i] == A[i + 2] - A[i + 1]:
                dp += 1
                res += dp
            else:
                dp = 0
        return res

File: 413_Arithmetic_Slices.cpp
class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& A) {
        if (A.size() < 3) return 0;
        int dp = 0, res = 0;
        for (int i = 2; i < A.size(); ++i) {
            if (A[i] - A[i - 1] == A[i - 1] - A[i - 2]) ++dp;
            else dp = 0;
            res += dp;
        }
        
        return res;
    }
};
